lib/SchemaMigrator.js

```diff
@@ -98,10 +107,55 @@ Attributes.prototype.migrate = function() {
  * Index definition migrations
  */
 function Index(parentMigrator, current, proposed) {
-    Unsupported.call(this, 'index', current, proposed);
+    var self = this;
+    self.current = current;
+    self.proposed = proposed;
+    self.currentSchema = parentMigrator.current;
+    self.proposedSchema = parentMigrator.proposed;
+
+    self.addIndex = proposed.filter(function(x) { return !self._hasSameIndex(self.current, x); });
```

Comment:
The way we build up all this state in the migrator is a bit ugly imho, but I don't see an easy way to avoid it with the current overall migrator design, either.